cmd/nvidia-dra-controller/imex.go

```diff
@@ -105,19 +107,29 @@ func (m *ImexManager) manageResourceSlices(ctx context.Context, owner resourcesl
 		return fmt.Errorf("error starting resource slice controller: %w", err)
 	}
 
+	imexChannelPool := make(map[string]int)
+
 	m.waitGroup.Add(1)
 	go func() {
 		defer m.waitGroup.Done()
 		for {
 			select {
 			case addedDomain := <-addedDomainsCh:
```

Comment:
Does it make sense to add a type for a domain so that we don't have to keep parsing this?

Reply:
I think is a good idea to consider moving forward, but a limitation is 

https://github.com/kubernetes/kubernetes/blob/d9b95ea94f41b942440adbdfdbe35fb78a2ed6f7/staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go#L77-L80

Maybe if the map was to interface instead of string, we could think about it

Reply:
I don't think its needed for a domain itself, but it probably is for the new imexChannelOffsets abstraction (which if done right, should be the only thing ever doing this parsing going forward).